tests/ingest/ingest_with_text_embedding_processor.yaml

```diff
@@ -0,0 +1,108 @@
+$schema: ../json_schemas/test_story.schema.yaml
+
+skip: false
+description: |
+  This test story checks that we can create an ingest pipeline with a text 
+  embedding processor a uses it to create an index.
+epilogues:
+  - path: /_ingest/pipeline/books_pipeline
+    method: DELETE
+    status: [200, 404]
+  - path: /books
+    method: DELETE
+    status: [200, 404]
+  # - path: /_plugins/_ml/model_groups/{model_group_id}
+  #   method: DELETE
+  #   status: [200, 404]
+  #   parameters_from_environment:
+  #     model_group_id: test_model_group_id
+  - path: /_plugins/_ml/models/{model_id}
+    method: DELETE
+    status: [200, 404]
+    parameters_from_environment:
+      model_id: test_model_id
+chapters:
+  - synopsis: Configure cluster
+    path: /_cluster/settings
+    method: PUT
+    request_body:
+      payload:
+        persistent:
+          plugins:
+            ml_commons:
+              only_run_on_ml_node: false
+  # - synopsis: Create model group
+  #   path: /_plugins/_ml/model_groups/_register
+  #   method: POST
+  #   request_body:
+  #     payload:
+  #       name: "NLP_Group"
+  #       description: "Model group for NLP models"
+  #   remember:
+  #     payload:
+  #       test_model_group_id: "model_group_id"
+  - synopsis: Create model
+    path: /_plugins/_ml/models/_register
+    method: POST
+    request_body:
+      payload:
+        name: huggingface/sentence-transformers/msmarco-distilbert-base-tas-b
+        version: "1.0.1"
+        model_format: TORCH_SCRIPT
+        # @TODO: Add model group, need to support passing things from the environment
+        # to the request body
+    remember:
+      payload:
+        model_registration_task_id: "task_id"
+  # - synopsis: Wait for model registration task to complete and keep the model id
+  #   path: /_plugins/_ml/tasks/{task_id}
+  #   method: GET
+  #   retry_on_response:
```

Comment:
I think we want to retry on any failure a number of times.

```
retry: 2
```

But I understand that in this case we are waiting for a field. Today we check for response schema. In addition I think we can add a `response_body` that expects some values, then `retry: 2` makes sense.

```
retry: 2
response_body:
    status: "CREATED"
```

Reply:
@miguel-vila I opened https://github.com/opensearch-project/opensearch-api-specification/issues/328 so we don't lose this. Comment if you're working on it!